boot/freeldr/freeldr/lib/fs/fat.c

```diff
@@ -1352,8 +1352,6 @@ BOOLEAN FatReadFile(PFAT_FILE_INFO FatFileInfo, ULONG BytesToRead, ULONG* BytesR
             *BytesRead += BytesToRead;
         }
         FatFileInfo->FilePointer += BytesToRead;
-        BytesToRead -= BytesToRead;
-        Buffer = (PVOID)((ULONG_PTR)Buffer + BytesToRead);
```

Comment:
You can also fix the associated comment above, into
```
// Now do the read and update BytesRead & FilePointer
```

Reply:
Right. Updated.